### PR TITLE
Feature/object crud

### PR DESCRIPTION
**Pull request by Peyton232**
Adds basic Creation, Update, and deactivation functionality for object type

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

### CODE REVIEW

File: highway/crud_object.go [hunk 1]
```diff
@@ -2,22 +2,158 @@ package highway
 
 import (
 	context "context"
+	"errors"
 
+	otv1 "github.com/sonr-io/blockchain/x/object/types"
```
Comment (joshLong145) [inline highway/crud_object.go]: rename to 'ot_v1'

File: highway/crud_object.go [hunk 2]
```diff
@@ -2,22 +2,158 @@ package highway
...
+	}
+
+	// Broadcast the message
+	res, err := s.cosmos.BroadcastUpdateObject(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	// decode otv1 fields to ot
+	outputFields := s.decodeObjectDocFields(res.WhatIs.ObjectDoc.Fields)
```
Comment (joshLong145) [inline highway/crud_object.go]: should we first check for non nill properties on result to prevent panics

File: highway/crud_object.go [hunk 3]
```diff
@@ -2,22 +2,158 @@ package highway
...
+}
+
+// -----------------
+// Helper Functions
+// -----------------
+
+// Translate ot (buf) fields to otv1 (blockchain) //TODO rename this
+func (s *HighwayServer) bufToBlockchain(bufFields []*ot.ObjectField) []*otv1.ObjectField {
+	var blockchainFields []*otv1.ObjectField
+	for _, v := range bufFields {
```
Comment (joshLong145) [inline highway/crud_object.go:142]: we should add some checks on the collection for before iterating over it.

File: highway/crud_object.go [hunk 4]
```diff
@@ -2,22 +2,158 @@ package highway
...
+		blockchainFields = append(blockchainFields, item)
+	}
+
+	return blockchainFields
+}
+
+// Translate the objectDoc otv1 (blockchain) fields to ot (buf) //TODO rename this
+func (s *HighwayServer) decodeObjectDocFields(blockchainFields map[string]*otv1.ObjectField) map[string]*ot.ObjectField {
+	// decode otv1 fields to ot
+	outputFields := make(map[string]*ot.ObjectField, len(blockchainFields))
```
Comment (joshLong145) [inline highway/crud_object.go:163]: check first for the length before making the new object

**Review by joshLong145 — CHANGES_REQUESTED** [review]

File: highway/crud_object.go (hunk 1 above)
Comment (Peyton232) [reply highway/crud_object.go]: Should I go ahead and change the name for registery, channels, and buckets to match this scheme?

Comment (joshLong145) [reply highway/crud_object.go]: if its low impact we can migrate naming in this PR

File: highway/crud_object.go (hunk 2 above)
Comment (Peyton232) [reply highway/crud_object.go]: Yes, cause len does not short circuit on nil pretty sure

File: highway/crud_object.go (hunk 3 above)
Comment (Peyton232) [reply highway/crud_object.go:142]: What checks are you thinking?

Comment (joshLong145) [reply highway/crud_object.go:142]: nill pointer check and length check

Comment (Peyton232) [reply highway/crud_object.go:142]: I believe these would both be redundant with the range function, it will simply not iterate on nil or 0 len.

**Review by joshLong145 — DISMISSED** [review]

File: highway/crud_object.go [hunk 5]
```diff
@@ -2,22 +2,171 @@ package highway
...
+		return nil
+	}
+
+	outputFields := make(map[string]*ot.ObjectField, len(blockchainFields))
+	for k, v := range blockchainFields {
+		outputFields[k] = &ot.ObjectField{
+			Label: v.GetLabel(),
+			Type:  ot.ObjectFieldType(v.GetType()),
+			Did:   v.GetDid(),
+			//Value:    v.Value,
```
Comment (mcjcloud) [inline highway/crud_object.go]: nit: remove commented code

File: highway/crud_object.go [hunk 6]
```diff
@@ -2,22 +2,171 @@ package highway
...
+
+	return blockchainFields
+}
+
+// Translate the objectDoc ot_v1 (blockchain) fields to ot (buf) //TODO rename this
+func (s *HighwayServer) decodeObjectDocFields(blockchainFields map[string]*ot_v1.ObjectField) map[string]*ot.ObjectField {
+	// decode ot_v1 fields to ot
+
+	if blockchainFields == nil || len(blockchainFields) < 1 {
+		return nil
```
Comment (mcjcloud) [inline highway/crud_object.go]: Returning an empty map would be better than `nil` here. There's not typically a need for nil maps.

**Review by mcjcloud — DISMISSED** [review]

File: highway/crud_object.go (hunk 6 above)
Comment (joshLong145) [reply highway/crud_object.go]: I would agree, keeps code paths nicer

File: highway/crud_object.go [hunk 7]
```diff
@@ -2,22 +2,221 @@ package highway
...
+			Type:        types.WhoIs_Type(regSess.Whois.GetType()),
+			Metadata:    regSess.Whois.GetMetadata(),
+			Timestamp:   regSess.Whois.GetTimestamp(),
+			IsActive:    regSess.Whois.GetIsActive(),
+		},
+		Credential: &types.Credential{
+			ID:              regSess.Credential.GetID(),
+			PublicKey:       regSess.Credential.GetPublicKey(),
+			AttestationType: regSess.Credential.GetAttestationType(),
+			// Authenticator: &types.Authenticator{  //TODO this causes nil dereference, figure out why
```
Comment (joshLong145) [inline highway/crud_object.go:196]: Authenticator needs to be hydrated, this object is critical

File: highway/crud_object.go (hunk 3 above)
Comment (joshLong145) [reply highway/crud_object.go:142]: we should guard so we can provide contextual information, perhaps it should return ([]*ot_v1.ObjectField, error)

**Review by joshLong145 — APPROVED** [review]
approving, noting that we should probably unify method signatures to return responses and errors in the future for better context

**Pull request merged**